### DISCUSSION TITLE
Intercepting the real route, not the 'as' route

### DISCUSSION THREAD

**Discussion by mihainsto** [vercel/next.js#21864] [Help]
Hello, I want to intercept with Router.events.on the real route, not the as route.

Let's say I have this link:

```
<Link href={'/thing'} as={'/completelyDifferentThing'}>
        click
 </Link>
```

And this intercepting method:

```
Router.events.on('routeChangeStart', (handler) => {
  console.log(handler)
})
```

This code logs 'completelyDifferentThing' not 'thing'. Why does this happen? shouldn't it log the real route not the 'as' route?

**Comment by leerob** [comment]
You don't need to use `as` anymore 😄 

> Optional decorator for the path that will be shown in the browser URL bar. Before Next.js 9.5.3 this was used for dynamic routes, check our previous docs to see how it worked.

https://nextjs.org/docs/api-reference/next/link

**Reply by mihainsto** [reply]
I have a case where I want to make a modal that shows a different link in the URL bar so it can be shared
